libbindgen/src/codegen/mod.rs

```diff
@@ -274,22 +269,28 @@ impl CodeGenerator for Module {
                item: &Item) {
         debug!("<Module as CodeGenerator>::codegen: item = {:?}", item);
 
-        if !ctx.options().enable_cxx_namespaces {
-            for child in self.children() {
-                ctx.resolve_item(*child)
-                    .codegen(ctx, result, whitelisted_items, &());
+        let codegen_self = |result: &mut CodegenResult| {
+            // FIXME: This could be less expensive, I guess.
```

Comment:
This is indeed pretty expensive. Any reason we can't continue to use `children()`?

```rust
let whitelisted_children = self.children()
    .iter()
    .filter(|id| whitelisted_items.contains_key(id));
for child in whitelisted_children {
    child.codegen(ctx, result, whitelisted_items, &());
}
```

Would like to understand this better before giving r+.